PORTING.md

```diff
@@ -0,0 +1,354 @@
+```
+
+### Compile a minimal example
+
+At this point you should test your changes to the build system and the port of the core driver.
+Create an empty example in `xpcc/examples/sam4/minimal`, containing:
+
+- the default `SConstruct` file:
+```python
+# path to the xpcc root directory
+xpccpath = '../../..'
+# execute the common SConstruct file
+execfile(xpccpath + '/scons/SConstruct')
+```
+
+- an empty `main.cpp` file:
+```cpp
+// your copyright with bsd license
+#include <xpcc/architecture/platform.hpp>
+
+// Leave this function empty for now
+extern "C" void xpcc_gpio_enable(void) {}
+
+int main()
+{
+    // Initialize one GPIO output using the datasheet.
+    // Don't forget to enable the GPIO clock!
+    // GPIO_PORTA_OUTPUT = (1 << 0);
+
+    while(1)
+    {
+        // Toggle the pin manually
+        // GPIO_PORTA_TOGGLE = (1 << 0);
+
+        // Create a very crude busy wait. The number don't have to
+        // be super accurate, just in the ball-park of a 1s delay.
+        // Use the initial CPU frequency in MHz here.
+        constexpr uint32_t fcpu_MHz = 8;
+        for (volatile uint32_t ii = 0; ii < fcpu_MHz * 250000; ii++) ;
+    }
+}
+```
+
+- the `project.cfg` file:
+```ini
+[build]
+device = sam4lc4b
+buildpath = ${xpccpath}/build/sam4/${name}
+```
+
+To compile this example, `cd` into the folder and execute `scons`.
+If everything goes well, there will be warnings about the missing `clock` and `gpio` drivers, but at the end there should be a printout of the resource usage, similar to this (but with realistic numbers):
+```sh
+cd examples/sam4/minimal
+scons
+[...]
+Memory Usage
+------------
+Device: sam4lc4b
+
+Program:    1000 bytes (1.0% used)
+(.data + .reset + .rodata + .text)
+
+Data:       1000 bytes (2.0% used) = 1 bytes static (1.0%) + 1000 bytes stack (1.0%)
+(.bss + .data + .noinit + .stack)
+
+Heap:      9999 bytes (97.0% available)
+(.heap1)
+```
+
+You can explore the generated code for the core driver in `xpcc/build/sam4/minimal/libxpcc/generated_platform/driver/core/cortex/`.
+
+If your development board is natively supported via OpenOCD (any of the scripts inside `openocd/share/openocd/scripts/`), you can just add it to your `project.cfg` file:
+```ini
+[openocd]
+configfile = board/atmel_sam4l8_xplained_pro.cfg
+```
+For custom setups, [read this](http://xpcc.io/reference/build-system/#project-configuration).
+
+You should now be able to program the device using `scons program`.
+Check if the LED is blinking at a constant interval.
+
+If the LED is not blinking and you have OpenOCD connected, you can [launch a GDB debug session](https://github.com/roboterclubaachen/xpcc/tree/develop/examples/stm32f3_discovery/gdb).
+
+If something goes wrong in the building step, check the previous steps, maybe you missed something, or we forgot to add something to this guide.
+If you're totally stuck write us an email (try to work the problem first though ;-).
```

Comment:
Include link to mailing list.
Also I think the last part could be understood to be condescending and should be clear to anyone using this guide. I'd just drop it.